tests/components/azure_event_hub/conftest.py

```diff
@@ -0,0 +1,126 @@
+"""Test fixtures for AEH."""
+from dataclasses import dataclass
+import logging
+from unittest.mock import MagicMock, patch
+
+from azure.eventhub.aio import EventHubProducerClient
+import pytest
+
+from homeassistant.components.azure_event_hub.const import DATA_FILTER, DATA_HUB, DOMAIN
+from homeassistant.config_entries import ConfigEntryState
+from homeassistant.const import STATE_ON
+from homeassistant.helpers.entityfilter import FILTER_SCHEMA
+
+from .const import AZURE_EVENT_HUB_PATH, BASIC_OPTIONS, PRODUCER_PATH, SAS_CONFIG_FULL
+
+from tests.common import MockConfigEntry
+
+_LOGGER = logging.getLogger(__name__)
+
+
+# fixtures for both init and config flow tests
+@pytest.fixture(autouse=True, name="mock_get_eventhub_properties")
+def mock_get_eventhub_properties_fixture():
+    """Mock azure event hub properties, used to test the connection."""
+    with patch(f"{PRODUCER_PATH}.get_eventhub_properties") as get_eventhub_properties:
+        yield get_eventhub_properties
+
+
+@pytest.fixture(name="filter_schema")
+def mock_filter_schema():
+    """Return an empty filter."""
+    yield {}
+
+
+@pytest.fixture(name="entry")
+async def mock_entry_fixture(hass, filter_schema, mock_create_batch):
+    """Create the setup in HA."""
+    entry = MockConfigEntry(
+        domain=DOMAIN,
+        data=SAS_CONFIG_FULL,
+        title="test-instance",
+        options=BASIC_OPTIONS,
+    )
+    hass.data[DOMAIN] = {DATA_FILTER: FILTER_SCHEMA(filter_schema)}
```

Comment:
Don't set or access anything in hass.data in the tests. Use `async_setup_component` to set up the integration, passing the whole config of Home Assistant. First add the config entry to hass. Then set up the integration with `async_setup_component`. This will first call `async_setup` of the integration with the config and then call `async_setup_entry` of the integration with the config entry.